src/day_two/mod.rs

```diff
@@ -0,0 +1,106 @@
+// https://adventofcode.com/2021/day/2
+use std::str::FromStr;
+
+enum Direction {
+    Forward,
+    Up,
+    Down,
+}
+
+impl FromStr for Direction {
+    type Err = std::io::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "forward" => Ok(Direction::Forward),
+            "up" => Ok(Direction::Up),
+            "down" => Ok(Direction::Down),
+            _ => Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                format!("{} is an invalid direction", s),
+            )),
+        }
+    }
+}
+
+struct Command {
+    direction: Direction,
+    distance: u32,
+}
+
+// https://fettblog.eu/rust-enums-wrapping-errors/
+enum CommandParseError {
```

Comment:
I'm not too familiar with rust conventions but I personally feel like you could move this and its trait impls to a separate file (errors.rs or something).

Reply:
Just read your comment above. We'll worry about this later.